contracts/proxies/TokenManagerProxy.sol

```diff
@@ -28,8 +28,8 @@ contract TokenManagerProxy is ITokenManagerProxy {
         tokenId = tokenId_;
         address impl = _getImplementation(IInterchainTokenService(interchainTokenServiceAddress_), implementationType_);
 
-        (bool success, ) = impl.delegatecall(abi.encodeWithSelector(TokenManagerProxy.setup.selector, params));
-        if (!success) revert SetupFailed();
+        (bool success, bytes memory error) = impl.delegatecall(abi.encodeWithSelector(TokenManagerProxy.setup.selector, params));
```

Comment:
returnData instead of error? Or change the event attribute above for consistency